tools/svsm/Launch_SVSM.sh

```diff
@@ -0,0 +1,596 @@
+    -o ConnectTimeout=1 \
+    "${1}" "${2}"
+}
+
+verify_snp_guest() {
+  # Exit if SSH private key does not exist
+  if [ ! -f "${GUEST_SSH_KEY_PATH}" ]; then
+    >&2 echo -e "SSH key not present [${GUEST_SSH_KEY_PATH}], cannot verify guest SNP enabled"
+    return 1
+  fi
+
+  # Look for SNP enabled in guest dmesg output
+  local snp_dmesg_grep_text="Memory Encryption Features active:.*SEV-SNP"
+  local snp_enabled=$(ssh_guest_command "sudo dmesg | grep \"${snp_dmesg_grep_text}\"")
+
+  [[ -n "${snp_enabled}" ]] \
+    && { echo "DMESG REPORT: ${snp_enabled}"; echo -e "SNP is Enabled"; } \
+    || { >&2 echo -e "SNP is NOT Enabled"; return 1; }
+}
+
+wait_and_verify_snp_guest() {
+  local max_tries=30
+  
+  for ((i=1; i<=${max_tries}; i++)); do
+    if ! (verify_snp_guest >/dev/null 2>&1); then
+      sleep 1
+      continue
+    fi
+    verify_snp_guest
+    return 0
+  done
+  
+  >&2 echo -e "ERROR: Timed out trying to connect to guest"
+  return 1
+}
+
+wait_and_retry_command() {
+  local command="${1}"
+  local max_tries=30
+  
+  for ((i=1; i<=${max_tries}; i++)); do
+    if ! (${command} >/dev/null 2>&1); then
+      sleep 1
+      continue
+    fi
+    ${command}
+    return 0
+  done
+  
+  >&2 echo -e "ERROR: Timed out trying to connect to guest"
+  return 1
+}
+setup_guest_attestation() {
+  ssh_guest_command "
+   
+    # Update and install necessary packages
+    echo 'Updating package list...'
+    sudo apt-get update
+
+    echo 'Installing necessary packages...'
+    sudo apt-get install -y git build-essential libtss2-dev tpm2-tools
+
+    echo 'Installing Rust...'
+    source ""${HOME}"/.cargo/env" 2>/dev/null || true
+
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -sSf | sh -s -- -y
+    source ""${HOME}"/.cargo/env" 2>/dev/null
+    "    
+
+  # Check if 'snpguest' directory exists and remove it if it does
+  ssh_guest_command "
+    if [ -d \"snpguest\" ]; then
+        echo 'Directory snpguest exists. Removing it...'
+        rm -rf snpguest
+    else
+        echo 'Directory snpguest does not exist.'
+    fi
+"
+
+  # Clone the repository and build the project
+  ssh_guest_command "git clone https://github.com/virtee/snpguest.git"
+  ssh_guest_command "cd snpguest  &&  source \$HOME/.cargo/env &&  cargo build --release"
+}
+
+
+attest_guest() {
+  local cpu_code_name="genoa"  
+
+  # Request and display the snp attestation report with random data
+  ssh_guest_command "
+  cd $HOME/snpguest/target/release
+  sudo ./snpguest report attestation-report.bin request-data.txt --random --vmpl 3;
+  ./snpguest display report attestation-report.bin;
+  ./snpguest fetch ca pem ${cpu_code_name} .;
+  ./snpguest fetch vcek pem ${cpu_code_name} . attestation-report.bin;
+  ./snpguest verify certs .;
+  ./snpguest verify attestation . attestation-report.bin
+"
+}
+
+Verify_measurement(){
+
+    # Define the working directory and the output file
+   OUTPUT_FILE="$SETUP_WORKING_DIR/make_output.txt"
```

Comment:
Fix indentation

Reply:
Agreed. Thanks for pointing that out; I'll update it.